src/PluginInfo/PluginInfo.js

```diff
@@ -122,146 +133,177 @@ class PluginInfo {
-    //   </config>
-    //   <pods use-frameworks="true" inhibit-all-warnings="true">
-    //     <pod name="PromiseKit" />
-    //     <pod name="Foobar1" spec="~> 2.0.0" />
-    //     <pod name="Foobar2" git="git@github.com:hoge/foobar1.git" />
-    //     <pod name="Foobar3" git="git@github.com:hoge/foobar2.git" branch="next" />
-    //     <pod name="Foobar4" swift-version="4.1" />
-    //     <pod name="Foobar5" swift-version="3.0" />
-    //   </pods>
-    // </podspec>
+    /**
+     * <podspec>
+     *
+     * @example
+     *  <podspec>
+     *      <config>
+     *          <source url="https://github.com/brightcove/BrightcoveSpecs.git" />
+     *          <source url="https://github.com/CocoaPods/Specs.git"/>
+     *      </config>
+     *      <pods use-frameworks="true" inhibit-all-warnings="true">
+     *          <pod name="PromiseKit" />
+     *          <pod name="Foobar1" spec="~> 2.0.0" />
+     *          <pod name="Foobar2" git="git@github.com:hoge/foobar1.git" />
+     *          <pod name="Foobar3" git="git@github.com:hoge/foobar2.git" branch="next" />
+     *          <pod name="Foobar4" swift-version="4.1" />
+     *          <pod name="Foobar5" swift-version="3.0" />
+     *      </pods>
+     *  </podspec>
+     *
+     * @param {string} platform
+     */
     getPodSpecs (platform) {
-        return _getTagsInPlatform(this._et, 'podspec', platform, tag => {
-            let declarations = null;
-            let sources = null;
-            let libraries = null;
+        return this._getTagsInPlatform('podspec', platform).map(tag => {
             const config = tag.find('config');
             const pods = tag.find('pods');
-            if (config != null) {
-                sources = config.findall('source').map(t => ({
-                    url: t.attrib.url
-                })).reduce((acc, val) => {
-                    return Object.assign({}, acc, { [val.url]: { source: val.url } });
-                }, {});
-            }
-            if (pods != null) {
-                declarations = Object.keys(pods.attrib).reduce((acc, key) => {
-                    return pods.attrib[key] === undefined ? acc : Object.assign({}, acc, { [key]: pods.attrib[key] });
-                }, {});
-                libraries = pods.findall('pod').map(t => {
-                    return Object.keys(t.attrib).reduce((acc, key) => {
-                        return t.attrib[key] === undefined ? acc : Object.assign({}, acc, { [key]: t.attrib[key] });
-                    }, {});
-                }).reduce((acc, val) => {
-                    return Object.assign({}, acc, { [val.name]: val });
-                }, {});
-            }
+
+            const sources = config && config.findall('source')
+                .map(el => ({ source: el.attrib.url }))
+                .reduce((acc, val) => Object.assign(acc, { [val.source]: val }), {});
+
+            const declarations = pods && pods.attrib;
```

Comment:
The original performed `reduce` but here it is no longer doing `reduce`. Is this intended?

Reply:
This is intended, yes. https://github.com/apache/cordova-common/pull/132/commits/386566fcfffc9ac70076525a80283093b7936b42 removes the reduction which did filter out values that are `undefined`. However, I claim such values cannot exist in the attributes object of an `et.Element` unless manually added after parsing the XML.